_extras/guide.md

```diff
@@ -26,9 +26,13 @@ is included as [`code/gen_inflammation.py`](../code/gen_inflammation.py).
 
 ## Overall
 
-This lesson is written as an introduction to Python,
-but its real purpose is to introduce the single most important idea in programming:
-how to solve problems by building functions,
+This lesson is written as an introduction to Python, 
+and our goal is to explain to the students that all computer software 
+no matter what langauge they are written in is composed of concepts like 
+variables, conditionals, looping and functions.
+We make a special effort to introduce the functions, the single most important idea in programming
+that allows reusability and efficient logical organization of the software.
+We show how to solve problems by building functions,
```

Comment:
I wasn't fond of the original language because it is so strongly worded with saying "the single most important idea in programming". I like that you add in a reason for why functions are important. However, I think the new wording takes some of the focus off of problem solving and puts it more on concepts of programming, where I think we want to highlight that we are teaching learners to solve problems with code, not about the code itself. I suggest tweaking the language to put the focus back on problem solving while retaining your explanation of why functions are important. What do you (and anyone else who cares to share) think about:
```
This lesson is written as an introduction to Python,
but its real purpose is to introduce an important idea in programming:
how to solve problems by building functions,
units of code that allow reusability and efficient logical organization of software.
In order to teach that,
we must teach people a little about
programming concepts such as variables, conditionals, loops,
and the mechanics of manipulating data with lists and file I/O
so that their functions can do things they actually care about.
```

Reply:
I like your suggestions. Maybe "file I/O" can be a little intimidating?
How about the following:

```
This lesson is written as an introduction to Python,
but its wider purpose is to introduce an important idea in programming:
how to solve problems by building functions, units of code that allow
reusability and efficient logical organization of software.

So that our functions can do things we actually care about,
we cover programming concepts such as variables, conditionals,
loops, and the mechanics of manipulating data with lists,
and of reading from files and writing to files.
```

Reply:
I would agree with both points, to make a special emphasis on functions, and to perhaps not to intimidate the potentially non-technical crowds with terms like the file I/O operations. I like @slel's version, since it incorporates both improvements. I should note though, that the training did not have the feeling like there was a particular effort on the introduction and practice of functions. If that's the intent, perhaps the curriculum might need updating sometime as well.

Reply:
I agree with @slel (thanks for commenting!) about replacing the term "file I/O", though I don't think we really spend time on writing to files. Maybe the second sentence could be:
```
So that our functions can do things we actually care about,
we cover programming concepts such as variables, conditionals,
loops, lists, and the mechanics of manipulating data read from files.
```

Reply:
Very good, for the sake of simplicity I suggest: 
```suggestion
So that our functions can do things we actually care about,
we cover programming concepts such as variables, conditionals,
loops, lists, and reading data from files.
```